lab4/viewpoints/context/context.tex

```diff
@@ -1,8 +1,349 @@
 \section{Context viewpoint}
 
+
+\subsection{System Scope and Responsibilities}
+
+\subsection*{Scope}
+EduPal is a learning aid platform for students and teachers. It allows educators to upload course information, and students to access learning material and tools.
+
+\begin{itemize}
+    \item Pomodoro Timer: Helps students manage study sessions effectively.
+    \item Note Creation and Export: Allows students to create notes and export them in PDF format.
+    \item Goal Setting: Enables users to set and track educational goals.
+    \item ChatGPT Integration: Provides an AI-powered assistant to help answer questions and provide learning support.
+    \item Subject and Topic Management: Users can organize their studies by subjects and topics, and upload conspectuses.
+    \item Real-Time Quizzing: This will allow users to create quiz rooms, share access codes, and participate in real-time quizzes, with results displayed immediately.
+\end{itemize}
+
+
+\subsection{Identity of External Entities and Services and Data Used}
+
+\subsection*{External Entities}
+\begin{itemize}
+    \item \textbf{Users (Students and Teachers):} Interact with EduPal to upload, manage, and access course content or take quizzes, utilise the features such as Pomodoro timer, quizzes, etc.
+    \item \textbf{PostgreSQL Database:} The central data storage system for EduPal. Stores all application data, except conspectus files.
+    \item \textbf{ChatGPT API:} External service used for providing AI-based assistance and responses within the platform.
+\end{itemize}
+
+\subsection*{Data Used}
+\begin{itemize}
+    \item \textbf{User Data:} Includes personal information, preferences, learning progress, and quiz results.
+    \item \textbf{Course Content and upvotes:} Uploaded by users, includes any kind of conspectuses and conspectus ratings.
+    \item \textbf{Notes and Goals:} Created by users, includes topic notes, goal-setting data, and progress tracking.
+    \item \textbf{Quiz Data:} Includes questions, answers, and results of the real-time quizzes.
+    \item \textbf{ChatGPT Interaction Data:} Includes user queries and responses generated by ChatGPT.
+\end{itemize}
+
+Two versions for this concern
+
+\subsection{Nature and Characteristics of External Entities}
+
+\subsection*{Users}
+\begin{itemize}
+    \item Require a reliable and user-friendly interface.
+    \item Access the system from various devices and locations.
+    \item Depend on system availability and performance for effective learning.
+\end{itemize}
```

Comment:
I think shorter version is better in this case. After you trim down other text, I believe you could merge some subsections.